### ISSUE TITLE
[Core][RDF] RDF HC always pops same dungeon over and over

### ISSUE THREAD

**Issue by Exorcista** [Stormforge-gg/Bugtracker-Fordring#1838]
**Current behaviour**: Like you read, today i've done HOS who knows how many times, lost the count on how many times this dungeon poped, i know the workarrounds but seriously? where's the "random" aspect coded at?

**Expected behaviour**:  Not the same dungeon like 10 times in a row please

**Comment by Godmatik** [comment]
Was HoS the daily by chance?

**Comment by VirtualReign** [comment]
This is understandable, with lower population counts expect to be doing the daily dungeon a lot (especially outside prime time). everyone and their alts wants it.

**Comment by Exorcista** [comment]
> Was HoS the daily by chance?

yes, but that means we will get same hc over and over? doesn't sound logic to me

**Comment by Godmatik** [comment]
It's because a lot of people are queuing for that specific dungeon for the daily. As said above this is a low pop server - so you end up grouping up those individuals and it gives you the dungeon when you queue.

**Comment by Exorcista** [comment]
> It's because a lot of people are queuing for that specific dungeon for the daily. As said above this is a low pop server - so you end up grouping up those individuals and it gives you the dungeon when you queue.

so the code forces people with the quest in quest log to get "daily hc" as rdf hc drop everytime until everyone in the server completes it? that sounds so bad haha , time to queue specific i guess...

**Comment by Vishiswaz** [comment]
> > It's because a lot of people are queuing for that specific dungeon for the daily. As said above this is a low pop server - so you end up grouping up those individuals and it gives you the dungeon when you queue.
> 
> so the code forces people with the quest in quest log to get "daily hc" as rdf hc drop everytime until everyone in the server completes it? that sounds so bad haha , time to queue specific i guess...

No

When attempting to form a group, rdf will try to find a correct combination of 5 people who have all "selected" at least one instance in common. People who select Random essentially have "selected" all of them. But people who queue specific have only selected the one(s) they checkboxed.

So if rdf can make a group using 4 people queued random and 1 person who queued for one specific instance, it will do so, and that means the group will only be able to pop that one specific instance.

**Comment by Godmatik** [comment]
> > It's because a lot of people are queuing for that specific dungeon for the daily. As said above this is a low pop server - so you end up grouping up those individuals and it gives you the dungeon when you queue.
> 
> so the code forces people with the quest in quest log to get "daily hc" as rdf hc drop everytime until everyone in the server completes it? that sounds so bad haha , time to queue specific i guess...

No, as Vishi said above, those that queue Random heroic are placed in the same pool as those who queue for a specific dungeon. Obviously if someone in the 5 man group that is formed has only queued for HoS, then it will pick HoS.

**Comment by Exorcista** [comment]
all i understood is that u can force farm the rdf dungeon you need loot from, for example, glad to know that is allowed

**Comment by Godmatik** [comment]
Well, if you have a different suggestion on how it should work, I guess list it?

I suppose the only thing that would work around this is to have people queueing RDF only allowed to queue with others who also choose RDF and not those choosing single queue. I've never seen a Wrath server attempt this, as it would lead to massively long queue times for both parties. Also not applicable on a low pop server anyway.

**Comment by Exorcista** [comment]
> 

i think thats what i meant a queue for random hcs only and another for the "quest", ofc is longer queues since server is low pop but i guess is a workarround
